lib/dslash_ndeg_twisted_clover_preconditioned.cu

```diff
@@ -0,0 +1,133 @@
+#include <gauge_field.h>
+#include <color_spinor_field.h>
+#include <clover_field.h>
+#include <dslash.h>
+#include <worker.h>
+
+#include <dslash_policy.cuh>
+#include <kernels/dslash_ndeg_twisted_clover_preconditioned.cuh>
+
+/**
+   This is the gauged preconditioned twisted-clover operator 
+   acting on a non-degenerate quark doublet.
+*/
+
+namespace quda
+{
+  template <typename Arg> class NdegTwistedCloverPreconditioned : public Dslash<nDegTwistedCloverPreconditioned, Arg>
+    {
+      using Dslash = Dslash<nDegTwistedCloverPreconditioned, Arg>;
+      using Dslash::arg;
+      using Dslash::in;
+
+      unsigned int sharedBytesPerThread() const
+      {
+        return 2 * in.Ncolor() * 2 * sizeof(typename mapper<typename Arg::Float>::type);
+      }
+      
+    public:
+    NdegTwistedCloverPreconditioned(Arg &arg, const ColorSpinorField &out,
+                                    const ColorSpinorField &in) :
+      Dslash(arg, out, in)
+      {
+        TunableKernel3D::resizeVector(2, arg.nParity);
+        // this will force flavor to be contained in the block
```

Comment:
Is there a good way to enforce this with an `errorQuda` somewhere---perhaps by providing a custom tuning advance routine that just makes sure the `y` blocksize is always 2?

Reply:
I'm not sure entirely what is meant here.  The `resizeStep` method was introduced to achieve exactly this, e.g., enforce the block size is advanced in united of the step size = 2, and is what is used in other operators that need this locality (non-deg twisted mass, domain-wall M5, etc).  I don't really see a need for a custom routine when the general one does fine.

I'm not adverse to further customized additions though, but I'm not entirely convinced of the need for it here 😜

Reply:
Ah, fair enough, I hadn't noticed there was precedence for that approach (and it was literally the intended design)---I'm a bit paranoid about corner-cases, as you can probably tell. We can discuss it outside of the scope of this PR.